### ISSUE TITLE
Created default NetworkPolicies do not work when service port is different then target port

### ISSUE THREAD

**Issue by TomVgw** [application-stacks/runtime-component-operator#606]
### What did you do?

We have upgrade our OpenLibertyOperator.  Due to the upgrade, the operator started creating Network Policies.  
For the creation of the Network policies, it uses utils from the runtime-component-operator.
Creation is done [here](https://github.com/OpenLiberty/open-liberty-operator/blob/699fcaa253d30ece5b2c66d0f2ffc1d89806bd7b/controllers/openlibertyapplication_controller.go#L383), function definition [here](https://github.com/OpenLiberty/open-liberty-operator/blob/699fcaa253d30ece5b2c66d0f2ffc1d89806bd7b/controllers/openlibertydump_controller.go#L9).

The specs of the service that was created by the operator looks like this:
`spec:
  clusterIP: 10.141.199.72
  ipFamilies:
    - IPv4
  ports:
    - name: http
      protocol: TCP
      port: 80
      targetPort: 9080
    - name: https
      protocol: TCP
      port: 443
      targetPort: 9443
  internalTrafficPolicy: Cluster
  clusterIPs:
    - 10.141.199.72
  type: ClusterIP
  ipFamilyPolicy: SingleStack
  sessionAffinity: None
  selector:
    app.kubernetes.io/instance: organisatiestructuur-service-dev`

### What did you expect to see?

We expected the Networkpolicies to allow ingress traffic., so no impact to a route defined within the OLA.
Within the network policies being created, we expected to find the ports of the pods, not the ports of the service.

We expected the following  code:
`spec:
  podSelector:
    matchLabels:
      apps.openliberty.io/name: organisatiestructuur-service-dev
  ingress:
    - ports:
        - protocol: TCP
          port: 9080
        - protocol: TCP
          port: 94433
      from:
        - namespaceSelector:
            matchLabels:
              policy-group.network.openshift.io/ingress: ''
        - namespaceSelector:
            matchLabels:
              network.openshift.io/policy-group: ingress
        - podSelector:
            matchLabels:
              app.kubernetes.io/part-of: organisatiestructuur-service-dev
          namespaceSelector:
            matchLabels:
              kubernetes.io/metadata.name: system-development-dev
        - namespaceSelector:
            matchLabels:
              network.openshift.io/policy-group: monitoring
  policyTypes:
    - Ingress`

### What did you see instead?

The created network policy mentions the port of the service.  Therefor, the defined routes failed.  Created network policy rules:
`spec:
  podSelector:
    matchLabels:
      apps.openliberty.io/name: organisatiestructuur-service-dev
  ingress:
    - ports:
        - protocol: TCP
          port: 80
        - protocol: TCP
          port: 443
      from:
        - namespaceSelector:
            matchLabels:
              policy-group.network.openshift.io/ingress: ''
        - namespaceSelector:
            matchLabels:
              network.openshift.io/policy-group: ingress
        - podSelector:
            matchLabels:
              app.kubernetes.io/part-of: organisatiestructuur-service-dev
          namespaceSelector:
            matchLabels:
              kubernetes.io/metadata.name: system-development-dev
        - namespaceSelector:
            matchLabels:
              network.openshift.io/policy-group: monitoring
  policyTypes:
    - Ingress`

It seems the ports of the service are used in stead of the targeted ports (pod ports)

### Environment

Openshift version v4.12.46 and Open Liberty Operator v1.3.0

### Possible solution

[This](https://github.com/application-stacks/runtime-component-operator/blob/main/utils/utils.go#L449) function needs to be adjusted. In stead of ba.GetService().GetPort()) one should use ba.GetService().GetTargetPort())

- [x] RCO
- [x] WLO
- [x] OLO
- [ ] Tests
- [x] Docs

**Comment by TomVgw** [comment]
Better overview of code this time....

Details of the service that was created by the OLA;
```
spec: 
  clusterIP: 10.141.199.72
  ipFamilies:
    - IPv4
  ports:
    - name: http
      protocol: TCP
      port: 80
      targetPort: 9080
    - name: https
      protocol: TCP
      port: 443
      targetPort: 9443
  internalTrafficPolicy: Cluster
  clusterIPs:
    - 10.141.199.72
  type: ClusterIP
  ipFamilyPolicy: SingleStack
  sessionAffinity: None
  selector:
    app.kubernetes.io/instance: organisatiestructuur-service-dev
```

Expected Network policies:
```
spec:
  podSelector:
    matchLabels:
      apps.openliberty.io/name: organisatiestructuur-service-dev
  ingress:
    - ports:
        - protocol: TCP
          port: 9080
        - protocol: TCP
          port: 9443
      from:
        - namespaceSelector:
            matchLabels:
              policy-group.network.openshift.io/ingress: ''
        - namespaceSelector:
            matchLabels:
              network.openshift.io/policy-group: ingress
        - podSelector:
            matchLabels:
              app.kubernetes.io/part-of: organisatiestructuur-service-dev
          namespaceSelector:
            matchLabels:
              kubernetes.io/metadata.name: system-development-dev
        - namespaceSelector:
            matchLabels:
              network.openshift.io/policy-group: monitoring
  policyTypes:
    - Ingress
```

Created Network policies:
```
spec:
  podSelector:
    matchLabels:
      apps.openliberty.io/name: organisatiestructuur-service-dev
  ingress:
    - ports:
        - protocol: TCP
          port: 80
        - protocol: TCP
          port: 443
      from:
        - namespaceSelector:
            matchLabels:
              policy-group.network.openshift.io/ingress: ''
        - namespaceSelector:
            matchLabels:
              network.openshift.io/policy-group: ingress
        - podSelector:
            matchLabels:
              app.kubernetes.io/part-of: organisatiestructuur-service-dev
          namespaceSelector:
            matchLabels:
              kubernetes.io/metadata.name: system-development-dev
        - namespaceSelector:
            matchLabels:
              network.openshift.io/policy-group: monitoring
  policyTypes:
    - Ingress
```

**Comment by kabicin** [comment]
@leochr [tests](https://github.ibm.com/websphere/operators/pull/151) and [doc update](https://github.com/OpenLiberty/open-liberty-operator/pull/603) ready for review. For tests to pass, I see that the required RCO dependency is already picked up by the OLO/WLO 1.3.3 changes.

**Comment by leochr** [comment]
@kabicin Thank you. Merged the doc update and assigned the review of tests PR to Melissa. 

Please open an issue to update [WLO docs for network policy](https://www.ibm.com/docs/en/was-liberty/base?topic=operator-configuration-examples#cfg-r-commoncomp__cc-networkpol__title__1) with the same.

Note that the ID team has requested to open WLO doc issues in a different repo going-forward: https://github.ibm.com/websphere/liberty-docs/issues
Please tag issues with `Operator` label. Thank you.

**Comment by leochr** [comment]
@TomVgw This was included in the recent 1.3.3 release. Thank you for reporting this.
